lib/controllers/getTarball.js

```diff
@@ -4,27 +4,43 @@ function getTarball(respond, route, unpm) {
   var version = route.params.version
   var name = route.params.name
   var started = false
+  var getTarballAsync = unpm.Package.getTarballAsync
+  var getTarballSync = unpm.Package.getTarball
 
-  var tarballStream = unpm.Package.getTarball(name, version)
+  var tarballStream = null
 
-  tarballStream.on('error', function(err) {
-    if(started) {
-      return unpm.log.error(err)
-    }
+  if (typeof getTarballSync === 'function') {
+    // Use synchronous behavior for fetching tarball. We're assuming it's coming from disk
+    tarballStream = getTarballSync(name, version)
+    doneFnc()
+  } else if (typeof getTarballAsync === 'function') {
+    // Async handler. Will provide a done function when the tarball is ready
+    getTarballAsync(name, version, function (tball) {
+      tarballStream = tball
+      doneFnc()
+    })
+  }
 
-    if(err.code === 'ENOTFOUND' || err.code === 'ENOENT' || err.notFound) {
-      return notFound()
-    }
+  var doneFnc = function () {
```

Comment:
Tests are currently failing because we call `doneFnc`before the anonymous function here is assigned to the variable (so `doneFnc` would be undefined)

I usually prefer to use named function rather than assigning functions to variables.

```function doneFnc () {...}```